test/parallel/test-child-process-spawn-shell.js

```diff
@@ -24,7 +24,7 @@ const echo = cp.spawn('echo', ['foo'], {
 });
 let echoOutput = '';
 
-assert.strictEqual(echo.spawnargs[echo.spawnargs.length - 1].replace(/"/g, ''),
+assert.strictEqual(echo.spawnargs[echo.spawnargs.length - 1].replace(/['"]/g, ''),
```

Comment:
why is this change needed?

Reply:
When `shellEscape: true` (as it was defaulted to in the original version), single-quoting is used to implement escape on POSIX (Unix) sh. As a result, it became required to trim the argument of quotes too. (Why not selective quoting? Selective quoting conflicts with shell extensions and is overcomplicated when there is a silver bullet.)

Before this was only for Windows, as it had to always call `'cmd', '/d', '/s', '/c', '"echo foo"'` (+verbatim). Now we have Unix also doing `'sh', '/c', "echo 'foo'"`.

I should probably add a `shellEscape: true` or revert this change.